logging, net: add ASN from peers on logs

brunoerg opened the pull request:
When using `-asmap`, you can check the ASN assigned to the peers only with the RPC command `getpeerinfo` (check `mapped_as` field), however, it's not possible to check it in logs (e.g. see in logs the ASN of the peers when a new outbound peer has been connected). This PR includes the peers' ASN in debug output when using `-asmap`.
 
Obs: Open this primarily to chase some Concept ACK, I've been using this on my node to facilitate to track the peers' ASN especially when reading the logs.

brunoerg committed:
net: add `GetMappedAS` in `CConnman`

brunoerg committed:
logging: log ASN when using `-asmap`

When using `-asmap`, it will log the ASN from the peer on some logs (e.g. when a new outbound peer has been connected).

Ayush170-Future commented on src/net_processing.cpp:
```diff
@@ -3410,9 +3410,10 @@ void PeerManagerImpl::ProcessMessage(CNode& pfrom, const std::string& msg_type,
+            LogPrintf("New outbound peer connected: version: %d, blocks=%d, peer=%d%s%s (%s)\n",
                       pfrom.nVersion.load(), peer->m_starting_height,
                       pfrom.GetId(), (fLogIPs ? strprintf(", peeraddr=%s", pfrom.addr.ToStringAddrPort()) : ""),
+                      (log_asmap ? strprintf(", mapped_as=%d", m_connman.GetMappedAS(pfrom.addr)) : ""),
```
Yeah, that does make more sense. I don't see any harm in automatically logging the asmap information.

brunoerg replied:
Yeah, I think it would be simpler to remove the new flag I created and log it when using `-asmap`.

Ayush170-Future commented:
Concept ACK.

brunoerg commented:
Just updated the description and changed the approach to leave it simpler. Before I had created a flag `-logasmap` (similar to `-logips`), now I removed it and when using `-asmap` it will includes the peers' ASN in debug output.

brunoerg commented:
>  It looks like logasmap is still added in the last push, though.

Sorry it was a leftover, fixed it.

brunoerg commented:
> Would be nice to log this for inbound connections too, I guess the receive version message: is most suitable for that?

Yes, it would be great to have this for both ones, just updated it!

jamesob commented:
Big concept ACK. Will test next week (post-honeymoon).

Sjors commented:
Tested that 0076bed45eb2b42111fa3f4c95181393c685a42e works.

During review I noticed `fLogIPs` which is controlled by `-logips` which is off by default. This settings seems to be ignored in a bunch of places. I don't know what it's original purpose is. If the issue was peer privacy (which #3764 implies), then even though an ASN is also not an IP, we should probably still hide it when `-logips` is false.

Ayush170-Future commented:
I agree with @Sjors here. In my last review comment, I suggested the same idea. Though I'm not sure how the peer privacy issue is relevant here. But, it makes more sense to only log ASN when <code>fLogIPs</code> is already true.

Maybe changing the name of <code>-logips</code> flag to something like <code>-lognetinfo</code> will better reflect its use and avoid confusion, but that would require a release note. (just a school of thought)

Sjors commented:
Reading that original discussion I think the current code is fine. The main concern was exposing the origin of transactions in log files under the _default_ configuration. Since we don't log transaction relay by default (it's extremely noisy) we can safely log the ASN (or even IP) of a node when it connects.

We can clean up the documentation and/or behavior of `-logips` in another PR.

brunoerg commented:
> We can clean up the documentation and/or behavior of -logips in another PR.

For sure, I can work on that after this one.

jamesob commented:
ACK 0076bed45eb2b42111fa3f4c95181393c685a42e ([`jamesob/ackr/27412.1.brunoerg.logging_net_add_asn_from`](https://github.com/jamesob/bitcoin/tree/ackr/27412.1.brunoerg.logging_net_add_asn_from))

Reviewed code, built locally. Change looks good. Going to test with -asmap today.


<details><summary>Show signature data</summary>
<p>

```
-----BEGIN PGP SIGNED MESSAGE-----
Hash: SHA512

ACK 0076bed45eb2b42111fa3f4c95181393c685a42e ([`jamesob/ackr/27412.1.brunoerg.logging_net_add_asn_from`](https://github.com/jamesob/bitcoin/tree/ackr/27412.1.brunoerg.logging_net_add_asn_from))

Reviewed code, built locally. Change looks good. Going to test with -asmap today.

-----BEGIN PGP SIGNATURE-----

iQIzBAEBCgAdFiEEGNRVI1NPYuZCSIrGepNdrbLETwUFAmQ1m4YACgkQepNdrbLE
TwWHLg/9EbvTm+b4IWsdpytUtiN2Y/v8+hX9dVZooWnvdOw5Bviryj5jr7eaUvhf
UpOBnwH/CH52kBpSZf5qGmDJKRECThSGYKeSnGzuDUhcfOuT6vax6/DIdy7qWsva
xx+t6KvRVcnkAif0yNVtKdV9+mng+W6KqmC0AndeqZ5H7VK+d/5Dp+W8r0tFSJsk
Bh5pTOYfaH53BZdjzQ00WWQJ8ljahuS3uDgTSIIAyJd3rtMv+vsvBN9dy5PQoMdT
gPWarbthuT1lQ80on/zD8SwKOPTcfyBwZbv74i5fzv8M71O8OBpQUrUcgPkI8oHI
2mzIgE1PaoN6zlnEtVcFjVLJgztNphbH5pmmzlqDCoOxoQ5fz7ZRIl0ck4Cz+8vv
DlawDyf7yRefZL3CKbgLWjOzXUOz5yrytj/YBGH6KzCyFUiv11kQTx73AtMnKKLY
M7tP0/7tsWVgCzl1SU0pL443q7ncw6S46UAoEL7aWmLT6HQ8My0tzMXUlZ+jNqIl
RroZRMCFoXC1B5dKe7o65V3PgWVW3JZbRZv5a5X7wKA5WWr5M8UK9q5PAyzq4MVY
DW87ozXrApaFilR+51beS2CHyVV2ysXkdq1hasBS0y3FNcYP66CFB4uiQeiUhd7U
RHpXx/05dL0hzOR4GTJoI4bHFpgNY8TSfOfZMxGFDTiwfmRWyK4=
=Sq2n
-----END PGP SIGNATURE-----

```

</p></details>

achow101 commented:
ACK 0076bed45eb2b42111fa3f4c95181393c685a42e
